### PR TITLE
Remove AIP-44 from models/taskinstance

### PR DESCRIPTION
**Pull request by rawwar**
related to https://github.com/apache/airflow/issues/44436

- [x] ./airflow/models/taskinstance.py: _merge_ti
- [x]  ./airflow/models/taskinstance.py: _add_log
- [x]  ./airflow/models/taskinstance.py: _update_ti_heartbeat
- [x]  ./airflow/models/taskinstance.py: _xcom_pull
- [x]  ./airflow/models/taskinstance.py: _get_template_context
- [x]  ./airflow/models/taskinstance.py: _handle_failure
- [x]  ./airflow/models/taskinstance.py: _record_task_map_for_downstreams
- [x]  ./airflow/models/taskinstance.py: _update_rtif
- [x]  ./airflow/models/taskinstance.py: _defer_task
- [x]  ./airflow/models/taskinstance.py: _handle_reschedule
- [x]  ./airflow/models/taskinstance.py: get_task_instance
- [x]  ./airflow/models/taskinstance.py: _clear_xcom_data
- [x]  ./airflow/models/taskinstance.py: _set_state
- [x]  ./airflow/models/taskinstance.py: _get_dagrun
- [x]  ./airflow/models/taskinstance.py: _check_and_change_state_before_execution
- [x]  ./airflow/models/taskinstance.py: _register_asset_changes_int
- [x]  ./airflow/models/taskinstance.py: save_to_db

### CODE REVIEW

**Comment by rawwar** [comment]
Once https://github.com/apache/airflow/pull/44496 is merged, I need to pull changes and fix remaining failing tests

**Comment by potiuk** [comment]
> Once #44496 is merged, I need to pull changes and fix remaining failing tests

Merged ! Go ahead!

**Review by eladkal — APPROVED** [review]

File: providers/src/airflow/providers/edge/worker_api/routes/_v2_routes.py
Comment (jscheffl) [inline providers/src/airflow/providers/edge/worker_api/routes/_v2_routes.py:89]: This will break the edge worker. As comments above this is atm an intentional copy from Airflow 2.10 interface state.
For mypy to pass you need to add a type ignore (like with the other methods which are gone in parallel or wait until I have merged #44434 which just removes the pieces of code.

Comment (jscheffl) [reply providers/src/airflow/providers/edge/worker_api/routes/_v2_routes.py:89]: Now the PR is merged, so probably after re-base you can drop these changes and should not be in-the-way anymore

**Comment by potiuk** [comment]
Rebased and mered in #44540 .. Almost out-of the box after I rebased - and you are co-author @rawwar :)

**Pull request closed without merging**
